mlflow/server/js/src/experiment-tracking/components/ExperimentPage.js

```diff
@@ -95,31 +95,59 @@ export class ExperimentPage extends Component {
     this.detectNewRunsTimer = setInterval(() => this.detectNewRuns(), DETECT_NEW_RUNS_INTERVAL);
   }
 
-  componentDidUpdate(prevProps) {
-    this.maybeReloadData(prevProps);
+  componentDidUpdate(prevProps, prevState) {
+    this.maybeReloadData(prevProps, prevState);
+  }
+
+  /** Snapshots desired attributes of the component's current state in local storage. */
+  snapshotComponentState() {
+    const store = ExperimentPage.getLocalStore(this.props.experimentId);
+    store.saveComponentState(new ExperimentPagePersistedState(this.state.persistedState));
   }
 
   static getDerivedStateFromProps(props, state) {
+    const store = ExperimentPage.getLocalStore(props.experimentId);
     if (props.experimentId !== state.lastExperimentId) {
       return {
         persistedState:
           state.lastExperimentId === undefined
             ? state.persistedState
-            : new ExperimentPagePersistedState({ orderByKey: DEFAULT_ORDER_BY_KEY }).toJSON(),
+            : new ExperimentPagePersistedState({
+                ...store.loadComponentState(),
+              }).toJSON(),
         lastExperimentId: props.experimentId,
-        lifecycleFilter: LIFECYCLE_FILTER.ACTIVE,
         nextPageToken: null,
         getExperimentRequestId: getUUID(),
         searchRunsRequestId: getUUID(),
         ...PAGINATION_DEFAULT_STATE,
       };
+    } else if (
+      // react to url being changed by user or using the back button
+      props.location.search !== state.urlState &&
+      props.history.action === 'POP'
```

Comment:
Why do we care if its a `POP` or some navigation in the future that changes the URL state. We could just care if the URL changed from the previous state, we can just update the local state? What do you think?

Reply:
Tried for a while to get it to work like that, but ended up in the situation where the `isLoading` prop was always `true` (obscuring the view with a spinner and causing all sorts of problems). 

I can take another look. It would probably require the state handling functions to be changed to no longer use the `this.setState({})` function and instead use `updateUrlWithViewState()`. This would cause an url change, which will be loaded into the component state in `getDerivedStateFromProps`